Dockerfile

```diff
@@ -0,0 +1,78 @@
+# Build Stage: Build the application
+FROM rust:latest AS builder
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy the main Cargo.toml and Cargo.lock files
+COPY Cargo.toml Cargo.lock ./
+
+# Copy the Cargo.toml files from each crate
+COPY web/Cargo.toml web/Cargo.toml
+COPY service/Cargo.toml service/Cargo.toml
+COPY entity_api/Cargo.toml entity_api/Cargo.toml
+COPY entity/Cargo.toml entity/Cargo.toml
+COPY migration/Cargo.toml migration/Cargo.toml
+
+# Verify binaries are created
+RUN ls -la /app/
+
+# Fetch dependencies based on the Cargo.toml and Cargo.lock
+RUN cargo fetch 
```

Comment:
`cargo build` will also fetch the dependencies I believe. Just curious why you broke this into a separate step. For my own education.

Reply:
Yep you're correct. The reasoning for this order is due to how Container Image Layers are built. Running `RUN cargo fetch` early in the `Dockerfile` caches the dependencies separately from the app code so Docker won't re-download them unless the `Cargo.toml` changes. When you `COPY` the code *and then* run `cargo build --release`, Docker only rebuilds if the code changes, making builds faster. Since we're using a multi-stage build, this also helps keep the final image compact and secure. Let me know if that answers your question.